homework3/jenkins.postman_collection.json

```diff
@@ -0,0 +1,335 @@
+{
+	"info": {
+		"_postman_id": "bf4c07ae-39a5-45cb-8ab3-45fc87507246",
+		"name": "jenkins",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
+	},
+	"item": [
+		{
+			"name": "get suite",
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "7edad2c6-18fb-4e98-be5b-72cf525c1ca5",
+						"exec": [
+							"const body = pm.response.json();\r",
+							"const suiteName = `${pm.collectionVariables.get(\"name\")} - ${pm.collectionVariables.get(\"JiraProjectName\")}`;\r",
+							"const TempItem = _.find(body, item => item.name === suiteName);\r",
+							"\r",
+							"if (TempItem === undefined) {\r",
+							"    const Request = {url: `https://tr.a1qa.com/index.php?/api/v2/add_suite/${pm.collectionVariables.get(\"TRprojectID\")}`,\r",
+							"    method: 'POST',\r",
+							"    header: {\r",
+							"        'Content-Type': 'application/json',\r",
+							"        'Authorization': request.headers.authorization\r",
+							"          },\r",
+							"    body: {\r",
+							"        mode: 'raw',\r",
+							"        raw: JSON.stringify({\t\r",
+							"            \"name\": suiteName,\r",
+							"        \t\"description\": \"Test\"\r",
+							"        })\r",
+							"    }};\r",
+							"    \r",
+							"    pm.sendRequest(Request, function (err, res) {\r",
+							"        console.log(err ? err : res.json());\r",
+							"        pm.collectionVariables.set(\"SuiteID\", res.json().id);\r",
```

Comment:
You would try to set the variable even if error was returned upon the request execution. This might cause `TypeError: Cannot read property 'json' of undefined`